### ISSUE TITLE
Optimizing UX for home slider on mobile screen

### ISSUE THREAD

**Issue by mparvazi** [PrestaShop/hummingbird#170]
When the media screen is mobile, captions should be displayed in bottom of images:

<img src="https://user-images.githubusercontent.com/85633460/152317997-99fe60fb-c97f-44a0-9ef2-20eb3ba72336.png" width="300">

The caption font sizes should be calculated responsively:

For example:

```
.carousel .carousel-content .h2 {
    font-size: calc(32px + (64 - 32) * ((80vw - 480px) / (1920 - 480)));
}
```

**Comment by SharakPL** [comment]
I disagree. For one it's possible to change color of the slide caption eg. to white on dark images, when editing a slide so you would have to force theme's base text color to override it and make it visible on white background instead of an image. Besides it doesn't even look sliderish this way.

First of all there should be option to set minimal height of the slide. Then there should be another option to set the behaviour of the image (should it be full width and auto-height or stretch to minimal height with object-fit: cover). Text should remain on the image. This would give at least minimal control on slide's behaviour.

Next: dynamic font-size should be set with **clamp()**. It's already [widely supported](https://caniuse.com/css-math-functions)

**Comment by NeOMakinG** [comment]
FYI: https://www.figma.com/file/LfVl5leeSKcVUhSaYwhbtM/New-Theme?node-id=81%3A9640

Here are the mockups of the slider on mobile

I think most of what you said is related to the module itself @SharakPL right? And then the theme should adapt depending on the module settings, maybe this issue should be created on the main repository of PrestaShop then, at least partially

**Comment by mparvazi** [comment]
Home slider module is very simple and currently there is no options for responsive display in xsmall media.
In classic theme this problem fixed by override rules.

So there is two options:
1. Fix in ps_imageslider (It may causes conflicts with classic theme)
2. Adding override rules in theme-refacto (for small and xsmall media)

**Comment by mparvazi** [comment]
Some options for controlling image fit in ps_imageslider:
https://github.com/PrestaShop/PrestaShop/issues/27580

**Comment by NeOMakinG** [comment]
Yeah, will be cool :)

**Comment by mparvazi** [comment]
@SharakPL 

I didn't know about clamp() function!

It's damn good :+1:

**Comment by mparvazi** [comment]
@NeOMakinG 

The problem for adding new features in ps_imageslider is that it always will be checked with classic theme for QA.

**Comment by NeOMakinG** [comment]
> @SharakPL
> 
> I didn't know about clamp() function!
> 
> It's damn good 👍
> 
> Btw some mobile browsers doesn't support.
> 
> This code can somehow be used as fallback: `font-size: calc(32px + (64 - 32) * ((80vw - 480px) / (1920 - 480)));`

Maybe we could find a light polyfill, that would do the job too

**Comment by NeOMakinG** [comment]
> @NeOMakinG
> 
> The problem for adding new features in ps_imageslider is that it always will be checked with classic theme for QA.

Yeah, because the classic theme should also be compatible with the module at the moment :/

**Comment by SharakPL** [comment]
> > @SharakPL
> > 
> > I didn't know about clamp() function!
> > 
> > It's damn good 👍
> > 
> > Btw some mobile browsers doesn't support.
> > 
> > This code can somehow be used as fallback: `font-size: calc(32px + (64 - 32) * ((80vw - 480px) / (1920 - 480)));`
> 
> Maybe we could find a light polyfill, that would do the job too

I've been looking for it lately but only found a polyfill that changes clamp to min/max combination. It seems browser support for all 3 is the same so I'm not sure what's the usefulness of such polyfill.

@mparvazi these mobile browsers already have support issues with BS5, just like IE11 (eg. CSS variables), but I think whole point of using this version is to move forward and forget about non-modern browsers. Otherwise this new theme should stick with BS4, just stable this time 😉

**Comment by pawelkierklo** [comment]
Hello everyone,

I'd like to propose adding the option to import images specifically for mobile devices. When a user accesses the mobile view, the page would load images tailored for the mobile version. Additionally, I suggest implementing the ability to set resolutions for both desktop and mobile versions.

Form Edit/Create in BO ps_imageslider:

- Title
- Link
- Show select (none/desk/mob/both)

- Image desktop
- Header desktop
- Wysiwyg desktop

- Image mobile
- Header mobile
- Wysiwyg mobile


---- Settings ----
- Resolution on desktop
- Resolution on mobile
